src/webhooks/handleLifetimeRefunded.ts

```diff
@@ -4,31 +4,58 @@ import CacheService from '../services/cache.service';
 import { StorageService, updateUserTier } from '../services/storage.service';
 import { UsersService } from '../services/users.service';
 import { AppConfig } from '../config';
+import { TierNotFoundError, TiersService } from '../services/tiers.service';
+import { handleCancelPlan } from './utils/handleCancelPlan';
+import Stripe from 'stripe';
+import { PaymentService } from '../services/payment.service';
 
 export default async function handleLifetimeRefunded(
   storageService: StorageService,
   usersService: UsersService,
-  customerId: string,
+  charge: Stripe.Charge,
   cacheService: CacheService,
+  paymentsService: PaymentService,
   log: FastifyLoggerInstance,
+  tiersService: TiersService,
   config: AppConfig,
 ): Promise<void> {
+  const customerId = charge.customer as string;
+  const userEmail = charge.receipt_email;
+  const invoiceId = charge.invoice as string;
   const { uuid } = await usersService.findUserByCustomerID(customerId);
 
-  await usersService.updateUser(customerId, { lifetime: false });
+  const invoice = await paymentsService.getInvoiceLineItems(invoiceId);
+  const productId = invoice.data[0].price?.product as string;
 
   try {
     await cacheService.clearSubscription(customerId);
   } catch (err) {
     log.error(`Error in handleLifetimeRefunded after trying to clear ${customerId} subscription`);
   }
+
   try {
-    await updateUserTier(uuid, FREE_INDIVIDUAL_TIER, config);
-  } catch (err) {
-    log.error(`Error while updating user tier: uuid: ${uuid} `);
-    log.error(err);
-    throw err;
-  }
+    await handleCancelPlan({
+      customerId,
+      customerEmail: userEmail ?? '',
+      productId,
+      usersService,
+      tiersService,
+      log,
+    });
+  } catch (error) {
+    if (!(error instanceof TierNotFoundError)) {
+      throw error;
+    }
+    await usersService.updateUser(customerId, { lifetime: false });
 
-  return storageService.changeStorage(uuid, FREE_PLAN_BYTES_SPACE);
+    try {
+      await updateUserTier(uuid, FREE_INDIVIDUAL_TIER, config);
+    } catch (err) {
+      log.error(`Error while updating user tier: uuid: ${uuid} `);
```

Comment:
Log this message _together_ with the `err` object, as we receive thousands of logs, it these ones do not go together, there is an inherent risk of losing the trace by not making the error searchable nor relatable to the customer that is affected by it. 

For instance, imagine you are watching the logs: 
```
<SOME_TIMESTAMP>: Error while updating user tier: uuid: <some_uuid>
.... 2543 logs here
<SOME_TIMESTAMP>: Unhandled error 'Tier not found' at ....
```

How are you going to relate both by a log query? Although they are related, we cannot establish the relation, therefore the first log is not too helpful.